code/game/machinery/computer/supply.dm

```diff
@@ -1,3 +1,162 @@
+/datum/supply_order
+	var/ordernum
+	var/orderedby = null
+	var/comment = null
+
+/datum/controller/supply
+	//supply points
+	var/points = 5000
+	var/points_per_process = 0
+	var/points_per_slip = 2
+	var/points_per_crate = 5
+	var/points_per_platinum = 5 // 5 points per sheet
+	var/points_per_plasma = 5
+	var/centcom_message = ""
+	var/contraband = 0
+	var/hacked = 0
+	//control
+	var/ordernum
+	var/list/shoppinglist = list()
+	var/list/requestlist = list()
+	var/list/supply_packs = list()
+	//shuttle movement
+	var/movetime = 1200
+	var/datum/shuttle/ferry/supply/shuttle
+
+	New()
+		ordernum = rand(1,9000)
+
+		for(var/typepath in (typesof(/datum/supply_pack) - /datum/supply_pack))
+			var/datum/supply_pack/P = new typepath()
+			supply_packs[P.name] = P
+
+
+
+	// Supply shuttle ticker - handles supply point regeneration
+	// This is called by the process scheduler every thirty seconds
+	proc/process()
+		points += points_per_process
+
+	//To stop things being sent to centcomm which should not be sent to centcomm. Recursively checks for these types.
+	proc/forbidden_atoms_check(atom/A)
+		if(isliving(A))
+			return 1
```

Comment:
Boolean variables and return values should use TRUE and FALSE constans instead of 1 and 0. This comment applies to all PR.